enterprise/server/remote_execution/cgroup/cgroup.go

```diff
@@ -31,6 +32,134 @@ const (
 	cidPlaceholder = "{{.ContainerID}}"
 )
 
+// min/max weight values for cgroup2 weight-based control
+const (
+	minWeight = 1
+	maxWeight = 10_000
+)
+
+// Setup configures the cgroup at the given path with the given settings.
+// Any settings for cgroup controllers that aren't enabled are ignored.
+// IO limits are applied to the given block device, if specified.
+func Setup(ctx context.Context, path string, s *scpb.CgroupSettings, blockDevice *block_io.Device) error {
+	m, err := settingsMap(s, blockDevice)
+	if err != nil {
+		return err
+	}
+	enabledControllers, err := ParentEnabledControllers(path)
+	if err != nil {
+		return fmt.Errorf("read enabled controllers: %w", err)
+	}
+	for name, value := range m {
+		controller, _, _ := strings.Cut(name, ".")
+		if !enabledControllers[controller] {
+			log.CtxWarningf(ctx, "Skipping cgroup %q setting for disabled cgroup controller %q", name, controller)
+			continue
+		}
+		settingFilePath := filepath.Join(path, name)
+		if err := os.WriteFile(settingFilePath, []byte(value), 0); err != nil {
```

Comment:
What happens with os.WriteFile if we try to write invalid values? Does it return an error or ignore the write? If it returns an error, maybe we don't need our own error checking.

If we need to do our own error checking, maybe we should check that all values are non-negative.